.github/workflows/tests.yml

```diff
@@ -50,21 +70,35 @@ jobs:
       uses: actions/setup-python@v1
       with:
         python-version: ${{ matrix.python-version }}
+
+    - name: Cache python wheels
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: |
+          ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('setup.py', 'docs/requirements.txt') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-${{ matrix.python-version }}-
+          ${{ runner.os }}-pip-
+
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip
-        pip install -e .
-        pip install -r docs/requirements.txt
+        python -m pip install --upgrade pip setuptools wheel
+        python -m pip install -e .[coverage]
 
     # Build the docs
     - name: Build docs to store
       run: |
         export PATH="$HOME/miniconda/bin:$PATH"
         sphinx-build -b html docs/ docs/_build/html -W --keep-going
 
-    # Run tests
+    # Run tests under coverage
     - name: Run the tests
-      run: pytest
+      run: pytest --cov pydata_sphinx_theme --cov-report term-missing:skip-covered --cov-fail-under ${{ env.COVERAGE_THRESHOLD }}
+
+    - name: Upload coverage
+      if: ${{ always() }}
```

Comment:
What is the purpose of this if statement, assuming that `always` is always true?